src/engine/QueryExecutionTree.h

```diff
@@ -52,6 +53,8 @@ class QueryExecutionTree {
     MINUS = 20
   };
 
+  enum ExportSubFormat { CSV, TSV, BINARY };
```

Comment:
Use a scoped enum:
```
enum class ExportSubFormat {CSV, TSV, BINARY };
using enum ExportSubFormat;
```
With the `using enum` you still have access to this format directly as `QueryExecutionTree::CSV`,
but the `enum class` does not allow implicit conversion to int.

Even better: don't use `using enum` here, but (if needed) in the functions where you use it,
because `QueryExecutionTree::CSV` is logically strange. Or maybe even declare this enum outside of the `QueryExecutionTree` class.